src/main/scala/it/pagopa/interop/backendforfrontend/api/impl/EServicesApiServiceImpl.scala

```diff
@@ -395,6 +395,34 @@ final case class EServicesApiServiceImpl(
     }
 
   }
+  
+  override def updateEServiceDocumentById(
+    eServiceId: String,
+    descriptorId: String,
+    documentId: String,
+    updateEServiceDescriptorDocumentSeed: UpdateEServiceDescriptorDocumentSeed
+  )(implicit
+    contexts: Seq[(String, String)],
+    toEntityMarshallerEServiceDoc: ToEntityMarshaller[EServiceDoc],
+    toEntityMarshallerProblem: ToEntityMarshaller[Problem]
+  ): Route = {
+    val result: Future[EServiceDoc] = for {
+      eServiceDoc <- catalogProcessService.updateEServiceDocumentById(
+        eServiceId = eServiceId,
+        descriptorId = descriptorId,
+        documentId = documentId,
+        updateEServiceDescriptorDocumentSeed = updateEServiceDescriptorDocumentSeed.toProcess
+      )(contexts)
+    } yield eServiceDoc.toApi
```

Comment:
no need of for comprehension here
```suggestion
    val result: Future[EServiceDoc] = catalogProcessService.updateEServiceDocumentById(
        eServiceId = eServiceId,
        descriptorId = descriptorId,
        documentId = documentId,
        updateEServiceDescriptorDocumentSeed = updateEServiceDescriptorDocumentSeed.toProcess
      )(contexts)
      .map(_.toApi)
```